tests/functional/mesh/work_test.go

```diff
@@ -587,6 +607,140 @@ func TestWork(t *testing.T) {
 				t.Fatal(err)
 			}
 		})
+		t.Run(testGroup+"/work submit to a remote node, change mesh config, reload mesh", func(t *testing.T) {
+			t.Parallel()
+			if strings.Contains(t.Name(), "kube") {
+				checkSkipKube(t)
+			}
+			controllers, m, _ := workSetup(t.Name())
+			defer tearDown(controllers, m)
+			// submit work from node 2 to node 3
+			unitID, err := controllers["node2"].WorkSubmit("node3", "echosleeplong50")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// wait for 10 seconds, and check if the work is in pending state
+			ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+			err = controllers["node3"].AssertWorkPending(ctx, unitID)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// read the file size for the work
+			workStatus, err := controllers["node2"].GetWorkStatus(unitID)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// modify the mesh to break connections betweeen all the nodes
+			modifiedData := mesh.YamlData{}
+			modifiedData.Nodes = make(map[string]*mesh.YamlNode)
+
+			modifiedData.Nodes["node2"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef: []interface{}{
```

Comment:
does defining NodeDef here do anything? I thought modifyclimeshfromyaml just ignores NodeDef anyways

Reply:
I have changed this. It will be pushed in the next commit.